File: python_programmers_school/level 1/하샤드 수.py
def solution(x):
    s = sum([int(i) for i in str(x)])
    if x % s == 0:
        return True 
    else:
        return False
    
# 더 간단하게 작성하기
# def Harshad(n):
#     return n%sum(int(x) for x in str(n)) == 0     # 0이면 자동으로 True, 0이 아닐 경우에는 자동으로 False 반환

File: javascript_programmers_school/level 1/하샤드 수.js
function solution(x) {
  const sum = (x + "").split("").reduce((acc, cur) => acc + Number(cur), 0);
  return x % sum ? false : true;
}

// 숫자로 접근하는 방법이 더 빠르기는 하다.
function solution(x) {
  let num = x;
  let sum = 0;
  do {
    sum += x % 10;
    x = Math.floor(x / 10);
  } while (x > 0);

  return !(num % sum);
}
